### DISCUSSION TITLE
confused about multiple databases

### DISCUSSION THREAD

**Discussion by jeff-h** [spatie/laravel-medialibrary#3221] [Q&A] [answered]
So our Laravel app uses two databases: `app` and `resources`. Both store various models which will use laravel-medialibrary eg

* the `app` db stores `user` models and
* the `resources` db stores models for `story`, `quotes` etc

I've created the `media` db table in both databases.

I haven't overridden the default Media class, so that's still using the default connection (`app`). I don't see a way to have a second (custom) Media class for the resources models (eg `stories`).

Yet the following puts an entry into the `resources` db:

```php
Story::find(1)->addMedia('cake.jpg')->toMediaCollection('feature');
```

That's not a problem — it's actually what I'd hope for, I guess, but I'm confused whether I'm going about this the right way (particularly having the `media` db table in both databases)?

**Comment by AntonLugtenburg** [comment, marked as answer]
You can also have a main database that stores media models for different model on different databases,

that would be done like this

```
namespace App\Models;

use Spatie\MediaLibrary\MediaCollections\Models\Media as BaseMedia;

class Media extends BaseMedia
{
    protected $connection = 'app';
}
```

And set this Model as default media model in the media library config...

In that case, every model would store and retrieve it from you app database
